src/sentry/api/endpoints/project_details.py

```diff
@@ -580,5 +580,7 @@ def delete(self, request, project):
                     'model': type(project).__name__,
                 }
             )
+            deleted_project = Project.objects.get(id=project.id)
```

Comment:
Why do we need to do this? You already have the `project` object. And you're querying for it just by primary key again, so you're effectively just querying back the same thing.

Reply:
I guess, I'm asking why not just `project.rename_on_pending_deletion()` instead and skip this extra step.

Reply:
@mattrobenolt I use `project.save()` during the `renamed_on_pending_deletion()` call. Which the object referenced there does not have `status = PENDING_DELETE`, because above it uses `.filter` to update the project. I was more/less looking for the easiest way to change it. I _could_ have just done `project.save(fields=[whatever-the-fields-needed-to-rename-are])` but it seemed like it would be less confusing to just reference the updated project you actually want to save.

A better way to do this would have just been to refactor the existing code. Instead of using the filter update check above, check if the project you have already isn't pending deletion. And if it's not update it and then call `project.rename_on_pending_deletion()` I can do that instead.